src/python/ensembl/io/genomio/utils/archive_utils.py

```diff
@@ -16,14 +16,27 @@
 
 __all__ = ["SUPPORTED_ARCHIVE_FORMATS", "extract_file"]
 
+from contextlib import contextmanager
 import gzip
 from os import PathLike
 from pathlib import Path
 import shutil
+from typing import Generator, TextIO
 
 import argschema
 
 
+@contextmanager
+def open_gz_file(file_path: PathLike) -> Generator[TextIO, None, None]:
```

Comment:
are we sure we don't want to pass additional args in `*args` and  have `"rt"`  hardcoded?